### ISSUE TITLE
On a particular config, with a particular toolchain, struct interprets double literal 0 as 0x3800000000000000

### ISSUE THREAD

**Issue by ryannathans** [micropython/micropython#2109]
In python3.5

```
>>> struct.pack('>d', 0)
b'\x00\x00\x00\x00\x00\x00\x00\x00'
```

in micropython

```
>>> struct.pack('>d', 0)
b'8\x00\x00\x00\x00\x00\x00\x00'
```

and weirdly enough in micropython

```
>>> struct.unpack(">d", b'8\x00\x00\x00\x00\x00\x00\x00')
(0.0,)
```

little endian does the same thing (but in little endian...)

**Comment by dpgeorge** [comment]
What platform are you running uPy on?  It works ok for me on unix.

**Comment by ryannathans** [comment]
Running on STM32F439

**Comment by dpgeorge** [comment]
It's most likely a problem with the `__aeabi_f2d` function in lib/libm/math.c.  You might like to try fixing it.

**Comment by dpgeorge** [comment]
This is now fixed by 3bedff0b3cbb5ae9bc499f8ec0108c54f26fc87c
